pkg/server/commands/batch_check_command_test.go

```diff
@@ -254,6 +263,58 @@ func TestBatchCheckCommand(t *testing.T) {
 		// response should have 1 key per check
 		require.Equal(t, len(response), numChecks)
 	})
+
+	t.Run("uses_command_cache_to_resolve_dupe_checks", func(t *testing.T) {
+		mockCheckResolver := graph.NewMockCheckResolver(mockController)
+		cmd := NewBatchCheckCommand(ds, mockCheckResolver, ts)
+		numChecks := 5
+		totalChecks := numChecks * 2
+		var checks []*openfgav1.BatchCheckItem
+
+		justinTuple := &openfgav1.CheckRequestTupleKey{
+			Object:   "doc:doc1",
+			Relation: "viewer",
+			User:     "user:justin",
+		}
+
+		ewanTuple := &openfgav1.CheckRequestTupleKey{
+			Object:   "doc:doc1",
+			Relation: "viewer",
+			User:     "user:ewan",
+		}
+
+		// add (2 * numChecks) tuples, only two of which are unique
+		for i := 0; i < numChecks; i++ {
+			checkItems := []*openfgav1.BatchCheckItem{
+				{
+					TupleKey:      justinTuple,
+					CorrelationId: fmt.Sprintf("justin_id_%d", i),
+				},
+				{
+					TupleKey:      ewanTuple,
+					CorrelationId: fmt.Sprintf("ewan_id_%d", i),
+				},
+			}
+			checks = append(checks, checkItems...)
+		}
+
+		// The check resolver should only receive two distinct checks
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).
+			Times(2).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, meta, err := cmd.Execute(context.Background(), params)
+
+		require.NoError(t, err)
+		require.Len(t, result, totalChecks)
+		require.Equal(t, meta.DuplicateCheckCount, totalChecks-2)
```

Comment:
**Request:** Can you add assertions for `DuplicateCheckCount` for all other tests?